### PR TITLE
#35568 Zero warnings

### PR DESCRIPTION
**Pull request by svick**

### CODE REVIEW

File: eng/src/BuildMetalamaPatterns.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
         <GenerateDocumentationFile>true</GenerateDocumentationFile>
         <LangVersion>latest</LangVersion>
         <Nullable>enable</Nullable>
-        <NoWarn>SA0001;CS8002</NoWarn>
+        <NoWarn>SA0001;CS8002;NU1903;NU1904</NoWarn>
```
Comment (prochan2) [inline eng/src/BuildMetalamaPatterns.csproj:10]: Comment on what these warnings mean, why they are suppressed, and when (if ever) this suppression should be removed.

File: src/tests/Metalama.Patterns.Caching.UnitTests/Backends/Distributed/BaseDistributedCacheTests.cs [hunk 2]
```diff
@@ -116,7 +116,7 @@ await Task.WhenAll( itemRemovedEvents.Select( e => e.Task ) ).WithTimeout( _time
         }
     }
 
-    [Fact( Timeout = _timeout )]
+    [Fact]
```
Comment (prochan2) [inline src/tests/Metalama.Patterns.Caching.UnitTests/Backends/Distributed/BaseDistributedCacheTests.cs:119]: How is the timeout managed now?

File: src/Metalama.Patterns.Observability/Implementation/ClassicStrategy/ClassicObservableExpression.cs [hunk 3]
```diff
@@ -97,7 +97,7 @@ public ClassicObservableExpression(
                 false => InpcBaseHandling.NotApplicable,
 
                 true when this.IsRoot =>
-                    fieldOrProperty.DeclaringType == builder.CurrentType
+                    fieldOrProperty.DeclaringType.Equals( builder.CurrentType )
```
Comment (prochan2) [inline src/Metalama.Patterns.Observability/Implementation/ClassicStrategy/ClassicObservableExpression.cs:100]: What is the impact of these changes? If this is a bug, there should be tests.

**Review by prochan2 — CHANGES_REQUESTED** [review]

File: src/Metalama.Patterns.Observability/Implementation/ClassicStrategy/ClassicObservableExpression.cs (hunk 3 above)
Comment (svick) [reply src/Metalama.Patterns.Observability/Implementation/ClassicStrategy/ClassicObservableExpression.cs:100]: I don't know if this is actually a bug, but `Equals` should be safer.

**Review by prochan2 — APPROVED** [review]

**Pull request merged**